packages/angular/projects/clr-angular/src/data/datagrid/datagrid.ts

```diff
@@ -226,6 +226,13 @@ export class ClrDatagrid<T = any> implements AfterContentInit, AfterViewInit, On
         if (!this.items.smart) {
           this.items.all = this.rows.map((row: ClrDatagridRow<T>) => row.item);
         }
+        // Remove any projected rows from the displayedRows container
+        // Necessary with Ivy off. See https://github.com/vmware/clarity/issues/4692
+        for (let i = this._displayedRows.length - 1; i >= 0; i--) {
+          if (this._displayedRows.get(i).destroyed) {
+            this._displayedRows.detach(i);
```

Comment:
I think this was due to an Angular bug—something along the lines of this: https://github.com/angular/angular/issues/38648. 

When `ViewRef` destroys itself independently without going through `ViewContainerRef` as we do in our `WrappedRow`'s `OnDestroy` cycle, the `ViewContainerRef` still holds the reference to the `ViewRef`. And that results in out-of-sync views and `ViewContainerRef`.

Reply:
I think I started with the `ViewContainerRef.remove()` at first, but it was having some side effects. Then I saw that in other part of this same TS file we're using detach() and it worked better. But I'm not sure if I was doing the `.destroyed` check back then, so I'll give it a try, out of curiosity at least. 
Thanks for the detailed review, Shijir!

Reply:
You're right, though it probably gets reclaimed after being detached, using remove gives us finer level of control and better protection against memory leaks. Updated all 3 PRs to use remove()